### ISSUE TITLE
Battery indicator

### ISSUE THREAD

**Issue by EDario333** [Emotiv/community-sdk#164]
Hi there, is me again,

A few days ago (exactly, 18 days ago) I submitted the issue #156 (still open, and still I can't run it successfully)... 

Well, now my question is this: the battery indicator my Emotiv Xavier.Pure.EEG (v 3.4.1, build 6497bbf) shows only a fourth part of green color. Is that correct? As the Quick Start Guide says, I've already charged the headset about 4 hours (on "off" position) and everytime is the same result. Also, this could be a factor (¿Low battery?) for FC6 and T8 channels are dead?

Attached image shows this issue
![10](https://cloud.githubusercontent.com/assets/24981684/22306021/3cffcbd8-e303-11e6-855f-0a085809d4e0.png)

**Comment by shahalh** [comment]
I have the same issue with with an Epoc+ device battery level.
I had some problems with electrodes, after wetting them a lot and pressing the reference electrodes to the head the black ones became green.

**Comment by dop3mine** [comment]
IZ ZE BATTRY INSERTED?

**Comment by EDario333** [comment]
so, @shahalh is it an issue with the battery? Do we must to request our warranties? And @dop3mine what do you mean with the "battery inserted"? I mean, the battery is really inserted on device, isn't?

**Comment by hgoycoolea** [comment]
Hi i am a Scientist that uses this sdk from its Beginings. I have never had any errors with it just so minor adjustments on Configuration.  I offer help if anyone has any problems with community sdk.  U rock emotiv people,  amazing tech!!

**Comment by EDario333** [comment]
@hgoycoolea, do you read the battery issue? Is that correct? Only a fourth part of green color is displayed. Perhaps is not a big problem, like @shahalh says: "wetting them a lot and pressing the reference electrodes to the head the black ones became green" is the real solution, but for me it seems a little "perturbing" the battery indicator.

**Comment by hgoycoolea** [comment]
Hi i had this issue before with some people that has hair (i am bold).  Sometimes is a battery issue but i dont think so,  is an interference issue, you can solve this with option A lubrication as you said or you can open the diod and put a larger contact with a niddle Shape, i have done this my self because some subject with their hair is a bit of a problem. Last thing i must add that the battery on insight and epoc is actually good one and Easy to replace if is a potence problem. But i dont think it is because you have good quality of reception. Try with a diff Shape so they can actually touch the skin. (my personal solution)  hope it help,  if need en i can help you on doing that.  Big hug from Mendoza, Argentina 🇦🇷

**Comment by EDario333** [comment]
thanks a lot @hgoycoolea. Indeed I really try with one guy without hair, and the results were the same. I suppose that the option of improve the contact sounds good. I will try it and later I write about it.

Hugs, from Mexico 🇲🇽

**Comment by hgoycoolea** [comment]
bueno así es mejor, en ingles era diferente. Mira Darío, si ya probaste con un calvo como yo, podrías ver de mejorar el contacto de la terminal como lo hice yo, sino abrilo y fijate de medir el contacto electrico que llega, si tenes un oscilador es re simple sino avisame y te doy otra solución mas del lado ectronico. Avisame hermano hay poca documentacion sobre esta tecnologia y me pase meses resolviendo problemas que fueron desde bateria hasta contacto asi que en que te pueda ayudar, escribe nomas !! abrazo

**Comment by EDario333** [comment]
Muchísimas gracias, ¿supongo que su nombre es Hugo? Buscaré el tiempo para probar esas opciones, aunque mi fuerte no es la electrónica (me considero más Ingeniero de software, aunque no me he metido como quisiera en ello, pues trabajo de maestro "investigador" en una universidad y poco es el tiempo que me queda para dedicarme a ello) lo intentaré con unos estudiantes...

Un gusto conocerlo, y pues a ver si en algún momento le comparto un poco más mi idea. Ando buscando un doctorado en el área de Ciencias Cognitivas pero siento que a cierto nivel hay muchos juegos de intereses de las universidades, al menos por aquí en mi zona.

Saludos.

**Comment by hgoycoolea** [comment]
jeje no no mi nombre es hector, yo tengo un doctorado en robotica y trabajo independiente. si necesitas ayuda avisame con gusto te ayudo. 
Te mando un abrazo y la idea de hacer un doctorado en ciencia cognitivas esta buenisimo yo trabajo en la deteccion de patrones en niños con TEA asi que te puedo ayudar bastante ahi.

**Comment by shahalh** [comment]
Hey there,
Don't force me to use Google translate! 😁

I want to make clear the battery and the electrodes are two different issues.
Make sure they are very wet, wait a little and wet again.

Now, the battery is fine and lasts a long time, I think that the problem is with the software indications that show quarter constantly.

**Comment by EDario333** [comment]
Ok Hector, muchas gracias. Seguiré caminando la idea por aquí, madurando las cosas y pues de antemano muchísimas gracias por su apoyo. Espero volvamos a leernos pronto. Saludos.

**Comment by hgoycoolea** [comment]
hi shahalh, we know, don't worry the electronics of the device is really simple and work very good. i wrote that battery works very good and easy to replace :). 
is about the contact, hehe don't worry about translate es is just like en. XD

**Comment by RuVT** [comment]
Yey hay mas gente de habla hispana de lo que pensaba en el repo :laughing: saludos.

I used the Emotiv headset 6 months ago in a Unity project and never got the response in the full sensors, 2 always in black, but was probably cause the headset passed through many hands before me, away from that no battery problems and worked well with enough lubrication.

**Comment by MKnierim** [comment]
Hello everyone,

not sure if this is the same issue: I am running the Java examples of the community SDK, and am getting an unexpected reading on the battery level (by running HeadsetInformationLogger.java). The result for batteryLevel.value() always is -1, regardless of whether the headset was just fully charged or ran for a longer time. 

Would someone happen to know why that is? MaxBatteryLevel.value() returns 5, so I'd expect there to be some variation between 0 and 5 for the actual battery level and -1 to represent an error message. Is this a bug in the code or am I doing something wrong?

Best regards
Michael

**Comment by hoangphamemotiv** [comment]
This is a issue in SDK library and EPOC+ headset firmware. battery value is incorrect.  we are trying to fix it in next version of firmware. 
for issue: "always is -1" is strange . it should be happened sometime. and can fix it by turn on/off headset, unplugin the dongle/BTLE

**Comment by juniormairdrops** [comment]
Hello, I have the Emotiv Epoc ... It's been a while since I used it, I went to do tests with it, but it doesn't turn on ... I thought it was unloading the battery, when I put it to charge, turns the green led, turns red and erase it. I left for almost 4 days on USB but does not charge, I believe the battery was already ... I looked on the site but did not find for purchase, does anyone know where I can buy one that is compatible? Is it 680mAh, or something I can do not to burn my device?
Thank you.
